wallet/pskt/src/wasm/pskt.rs

```diff
@@ -384,4 +396,317 @@ impl PSKT {
             .map_err(|e| Error::custom(format!("Failed to extract transaction: {e}")))?;
         Ok(tx.tx.mass())
     }
+
+    /// Extracts all input addresses from the PSKT.
+    /// This is useful for figuring out which private keys are required for signing.
+    #[wasm_bindgen]
+    pub fn addresses(&self, network_id: &NetworkIdT) -> Result<Vec<Address>> {
+        let network_id = NetworkId::try_cast_from(network_id)?.into_owned();
+        let prefix: Prefix = network_id.into();
+
+        let state_guard = self.state();
+        let inner = match state_guard.as_ref().unwrap() {
+            State::NoOp(Some(inner)) => inner,
+            State::Creator(pskt) => pskt.deref(),
+            State::Constructor(pskt) => pskt.deref(),
+            State::Updater(pskt) => pskt.deref(),
+            State::Signer(pskt) => pskt.deref(),
+            State::Combiner(pskt) => pskt.deref(),
+            State::Finalizer(pskt) => pskt.deref(),
+            State::Extractor(pskt) => pskt.deref(),
+            _ => return Err(Error::Custom("PSKT is not initialized".to_string())),
+        };
+
+        let addresses = inner
+            .inputs
+            .iter()
+            .filter_map(|input| input.utxo_entry.as_ref())
+            .filter_map(|utxo_entry| extract_script_pub_key_address(&utxo_entry.script_public_key, prefix).ok())
+            .collect::<Vec<_>>();
+
+        Ok(addresses)
+    }
+
+    /// Sign the PSKT with the provided private keys.
+    /// The method will find the inputs corresponding to the private keys and sign them.
+    /// This method performs partial signing, so if no private keys are provided, it will
+    /// return an unmodified PSKT.
+    #[wasm_bindgen]
+    pub fn sign(&self, private_keys: PrivateKeyArrayT, network_type: &NetworkTypeT) -> Result<PSKT> {
```

Comment:
also for multisig or other purposes Input.bip32_derivations and global.xpubs may be checked to define what keys must be used. but i think its unrelevant to sign fn that must allow caller to pass any signature to any input